src/main/java/org/terasology/module/health/components/DamageResistComponent.java

```diff
@@ -2,10 +2,15 @@
 // SPDX-License-Identifier: Apache-2.0
 package org.terasology.module.health.components;
 
-import org.terasology.engine.entitySystem.Component;
+import org.terasology.gestalt.entitysystem.component.Component;
 
 import java.util.Map;
 
-public class DamageResistComponent implements Component {
+public class DamageResistComponent implements Component<DamageResistComponent> {
     public Map<String, Float> damageTypes;
+
+    @Override
+    public void copy(DamageResistComponent other) {
+        this.damageTypes = other.damageTypes;
```

Comment:
Same as for the engine PR: is this a deep copy? Should this rather be something like

```java
this.damageTypes.clear();
this.damageTypes.putAll(other.damageTypes);
```

Additionally, the map might not even be initialized here :see_no_evil: